src/FakeItEasy/Core/FakeManager.CancellationRule.cs

```diff
@@ -44,18 +44,26 @@ public override void Apply(IInterceptedFakeObjectCall fakeObjectCall)
                 }
                 else
                 {
-                    var token = GetCanceledTokens(fakeObjectCall).First();
-                    token.ThrowIfCancellationRequested();
+                    GetCanceledToken(fakeObjectCall)?.ThrowIfCancellationRequested();

```

Comment:
Seeing this makes me think of a possible improvement (unrelated to this PR). I we passed some kind of "context" object to `IsApplicableTo` and `Apply`, we could avoid doing the same work twice (in this case, finding the cancellation token). `IsApplicableTo` would save some information in the context, and `Apply` would get it from there. Note that it would be a breaking change, though, since `IFakeObjectCallRule` is public.

Reply:
Yeah. I was also thinking of this. I'd even had some ideas about making it not so breaking. Or at least less breaking. For now I was interested in reducing the cost when the rule isn't applicable (since most aren't), but there's no reason not to reduce the cost of a hit.